ipdata/cli.py

```diff
@@ -0,0 +1,239 @@
+import csv
+import json
+import os
+import sys
+from ipaddress import ip_address
+from pathlib import Path
+from sys import stderr, stdout
+
+import click
+
+if __name__ == '__main__':
+    from ipdata import IPData
+else:
+    from .ipdata import IPData
+
+
+class WrongAPIKey(Exception):
+    pass
+
+
+class IPAddressType(click.ParamType):
+    name = 'IP_Address'
+
+    def convert(self, value, param, ctx):
+        try:
+            return ip_address(value)
+        except:
+            self.fail(f'{value} is not valid IPv4 or IPv6 address')
+
+    def __str__(self) -> str:
+        return 'IP Address'
+
+
+@click.group(help='CLI for IPData API', invoke_without_command=True)
+@click.option('--api-key', required=False, default=None, help='IPData API Key')
+@click.pass_context
+def cli(ctx, api_key):
+    ctx.ensure_object(dict)
+    ctx.obj['api-key'] = get_and_check_api_key(api_key)
+    if ctx.invoked_subcommand is None:
+        print_ip_info(api_key)
+    else:
+        pass
+
+
+def get_api_key_path():
+    home = str(Path.home())
+    return os.path.join(home, '.ipdata')
+
+
+def get_api_key():
+    key_path = get_api_key_path()
+    if os.path.exists(key_path):
+        with open(key_path, 'r') as f:
+            for line in f:
+                if line:
+                    return line
+    else:
+        return None
+
+
+def get_and_check_api_key(api_key: str = None) -> str:
+    if api_key is None:
+        api_key = get_api_key()
+    if api_key is None:
+        print(f'Please specify IPData API Key', file=stderr)
+        raise WrongAPIKey
+    return api_key
+
+
+@cli.command()
+@click.pass_context
+def init(ctx):
+    api_key = ctx.obj['api-key']
+    key_path = get_api_key_path()
+
+    ipdata = IPData(api_key)
+    res = ipdata.lookup('8.8.8.8')
+    if res['status'] == 200:
+        existing_api_key = get_api_key()
+        if existing_api_key:
+            print(f'Warning: You already have an IPData API Key "{existing_api_key}" listed in {key_path}. '
+                  f'It will be overwritten with {api_key}',
+                  file=stderr)
+
+        with open(key_path, 'w') as f:
+            f.write(api_key)
+        print(f'New API Key is saved to {key_path}')
+    else:
+        print(f'Failed to check the API Key (Error: {res["status"]}): {res["message"]}',
+              file=stderr)
+
```

Comment:
I get this error when trying to set a new api key;

```
> ipdata init test
Usage: ipdata init [OPTIONS]
Try "ipdata init --help" for help.

Error: Got unexpected extra argument (test)
```

"test" is a valid API key.